### DISCUSSION TITLE
How to make a custom helper for template? [V5]

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#3453] [Help] [answered]
Hello Team,
How to make a custom helper for templates?

**Comment by Julien-R44** [comment, marked as answer]
Hello, looks at this : https://docs.adonisjs.com/guides/views/data-flow#globals

You have to define your helpers in an preloaded file so :
````
node ace make:prldfile ViewHelpers
````
And select "during HTTP server"

Your .adonisrc.json should contains this now :
```json
{
  "preloads": [
    "./start/routes",
    "./start/kernel",
    {
      "file": "./start/ViewHelpers",
      "environment": [
        "web"
      ]
    }
  ],
```

Now you can define your helper in start/ViewHelpers : 

```typescript
import View from '@ioc:Adonis/Core/View'

View.global('nl2br', function (text) {
  return text.replace(/([^>\r\n]?)(\r\n|\n\r|\r|\n)/g, '$1<br />$2')
})
```

In your template : 
```html
<p> {{{ nl2br(post.description) }}} </p>
```

**Reply by pk4all** [reply]
@Julien-R44 Thanks, your comment is more helpful for me.
